diff --git a/sites/all/modules/contrib/scald_soundcloud/scald_soundcloud_player.tpl.php b/sites/all/modules/contrib/scald_soundcloud/scald_soundcloud_player.tpl.php
index a9440e3b..95b1dcb3 100644
--- a/sites/all/modules/contrib/scald_soundcloud/scald_soundcloud_player.tpl.php
+++ b/sites/all/modules/contrib/scald_soundcloud/scald_soundcloud_player.tpl.php
@@ -4,4 +4,4 @@
  *   Default theme implementation for the Scald Soundcloud player.
  */
 ?>
-<iframe width="100%" height="<?php print $vars['iframe_height'] ?>" scrolling="no" frameborder="no" src="http://w.soundcloud.com/player/?url=http://api.soundcloud.com/<?php print $vars['id'] ?>&amp;show_artwork=true"></iframe>
+<iframe width="100%" height="<?php print $vars['iframe_height'] ?>" scrolling="no" frameborder="no" src="https://w.soundcloud.com/player/?url=http://api.soundcloud.com/<?php print $vars['id'] ?>&amp;show_artwork=true"></iframe>
